### PR TITLE
Add client functions to allow integrations within other CLIs.

### PR DESCRIPTION
**Pull request by calavera**
This is a simplification of how the docker engine implements
this feature, but it will be ported there once this is merged.

Signed-off-by: David Calavera <david.calavera@gmail.com>

### CODE REVIEW

**Comment by vdemeester** [comment]
LGTM :angel:

**Comment by runcom** [comment]
LGTM

**Comment by runcom** [comment]
I cannot merge though :(

**Comment by LK4D4** [comment]
I wonder who can merge it :)
Maybe @icecrime

File: client/client.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+func Get(program ProgramFunc, serverURL string) (*credentials.Credentials, error) {
+	cmd := program("get")
+	cmd.Input(strings.NewReader(serverURL))
+
+	out, err := cmd.Output()
+	if err != nil {
+		t := strings.TrimSpace(string(out))
+
+		// do not return an error if the credentials are not
+		// in the keyckain. Let the caller ask for new credentials.
```
Comment (runcom) [inline client/client.go]: cross-posting from docker/docker: this comment isn't right anymore

Comment (calavera) [inline client/client.go]: removed, thanks!

**Comment by calavera** [comment]
> I wonder who can merge it :)

You should be able to merge it now :tada:

**Pull request merged**